### PR TITLE
chore: add support to generate sonar coverage reports

### PR DESCRIPTION
**Pull request by kranthie-sap**
* Add `jest-sonar` dependency.
* Update `jest` config in all modules to use new reporter to generate jest coverage reports that Sonar Cloud can use to publish code coverage metrics.
* Add sonar cloud configuration file

### CODE REVIEW

**Review by FredAtWork — DISMISSED** [review]

**Review by tobiasqueck — APPROVED** [review]

**Review by tobiasqueck — DISMISSED** [review]

File: packages/fe-fpm-writer/test/unit/section.test.ts [hunk 1]
```diff
@@ -136,7 +136,7 @@ describe('CustomSection', () => {
             expect(fs.read(fragmentPath)).toMatchSnapshot();
         });
 
-        const testVersions = [1.9, 1.85, 1.84,  1.86, 1.90];
+        const testVersions = [1.9, 1.85, 1.84, 1.86, 1.9];
```
Comment (devinea) [inline packages/fe-fpm-writer/test/unit/section.test.ts:139]: Why do we have `1.9` twice?

File: packages/fiori-freestyle-writer/jest.config.js [hunk 2]
```diff
@@ -3,7 +3,16 @@ module.exports = {
         '^.+\\.ts$': 'ts-jest'
     },
     collectCoverage: true,
-    coverageReporters: ['text','html'],
     collectCoverageFrom: ['src/**/*.ts'],
+    reporters: [
+        'default',
```
Comment (devinea) [inline packages/fiori-freestyle-writer/jest.config.js:8]: I assume `default`  covers  `coverageReporters: ['text','html'],` ?

File: sonar-project.properties [hunk 3]
```diff
@@ -0,0 +1,21 @@
+sonar.organization=sap-1
+sonar.projectKey=SAP_open-ux-tools
+sonar.sources=.
+sonar.inclusions=**/*.ts
+sonar.exclusions=**/*.test.ts
```
Comment (devinea) [inline sonar-project.properties:5]:
```suggestion
sonar.exclusions=**/*.test.ts, **/templates/**
```
I think the templates should be excluded. Is this the place to configure it?

File: packages/fe-fpm-writer/test/unit/section.test.ts (hunk 1 above)
Comment (tobiasqueck) [reply packages/fe-fpm-writer/test/unit/section.test.ts:139]: Why is this change in this PR? This is a regression currently in #269 which @stefanschreck is going to fix before merging. It should not be here ...

Comment (kranthie-sap) [reply packages/fe-fpm-writer/test/unit/section.test.ts:139]: @tobiasqueck: It looks like it was updated by prettier in the git hook. I'll undo this change.

File: sonar-project.properties (hunk 3 above)
Comment (kranthie-sap) [reply sonar-project.properties]: @devinea: Yes, I added this in the latest commit.

File: packages/fiori-freestyle-writer/jest.config.js (hunk 2 above)
Comment (kranthie-sap) [reply packages/fiori-freestyle-writer/jest.config.js:8]: @devinea: I couldn't find docs on the default reporters of jest. But, I did verify that the coverage is shown in the console and the html files are in the `coverage/lcov-report` folder.

**Review by devinea — DISMISSED** [review]
Thanks @kranthie-sap

**Review by tobiasqueck — DISMISSED** [review]

**Review by tobiasqueck — APPROVED** [review]

**Review by stefanschreck — APPROVED** [review]

**Pull request merged**